### PR TITLE
k3s-ansible submodule & removed k8s-vanilla and NGINX support

### PR DESCRIPTION
**Pull request by giubacc**
This PR is mainly a quality iteration for the K3s environment.

Highlights:

- Refactored Vagrantfile
- Added Support for OpenSuse/Leap Vagrant Box
- Added [k3s-ansible](https://github.com/k3s-io/k3s-ansible) submodule, this permits to create a 1 master and N workers.
- Removed support for vanilla-k8s and NGINX
- General improvements to Ansible playbooks: removed most of raw `command` statements in favor of specific modules.
- Cleaning

Fixes: https://github.com/aquarist-labs/s3gw-core/issues/62
Signed-off-by: Giuseppe Baccini <giuseppe.baccini@suse.com>

### CODE REVIEW

File: env/README.bm.md [hunk 1]
```diff
@@ -33,7 +33,7 @@ system. Additionally, it will deploy Longhorn and the s3gw in the cluster.
 
 ```
 $ cd ~/git/s3gw-core/env
-$ ./setup-k3s.sh
+$ ./setup.sh
```
Comment (m-ildefons) [inline env/README.bm.md:36]: This does no exist, is it still called setup-k3s.sh?

File: env/generate-spec.sh [hunk 2]
```diff
@@ -19,8 +19,8 @@ is_dev_env=false
 
 s3gw_image="ghcr.io/aquarist-labs/s3gw:latest"
 s3gw_image_pull_policy="Always"
-
-ingress="traefik"
+s3gw_ui_image="localhost/s3gw-ui:latest"
```
Comment (m-ildefons) [inline env/generate-spec.sh:22]: Half of what this script and the ansible roles below do could be handled by a simple `helm install` command, now that we have a helm chart.

**Review by m-ildefons — COMMENTED** [review]
Generally approved.
I just wonder how much sense there is to maintaining setup scripts for K8s clusters and ansible roles to deploy stuff.
Would it not make more sense to assume an existing K8s cluster of some sort, and just use helm to deploy the containers on it?
That would leave the setup of the cluster as an exercise to the user, but about half of what these scripts do is already covered by helm anyways. And the other half serves to set up a cluster in the first place.

The other problem I have with this is that we don't really expect our users to fish out the manifests from this repo anyways, when they want to deploy to their cluster, right?

**Comment by giubacc** [comment]
> Generally approved. I just wonder how much sense there is to maintaining setup scripts for K8s clusters and ansible roles to deploy stuff. Would it not make more sense to assume an existing K8s cluster of some sort, and just use helm to deploy the containers on it? That would leave the setup of the cluster as an exercise to the user, but about half of what these scripts do is already covered by helm anyways. And the other half serves to set up a cluster in the first place.
> 
> The other problem I have with this is that we don't really expect our users to fish out the manifests from this repo anyways, when they want to deploy to their cluster, right?

1)

Personally I see no problem at all assuming the user already has a kubernetes cluster set up from his own.
In this case he has just to use the helm chart and he lives happy.
Installing kubernetes from scratch, btw, has been the first attempt we made because we really didn't know the instrument and we had to learn it.
Said this, having a script/ansible or whatever toolery for replicating an environment quickly (minutes magnitude) could be, IMO, an added value to the overall project. 
This especially if developers wish to have a replicated common environment to work with.
Also, this could be a facilitation for the casual user, that perhaps, wouldn't care to exercise installing a kubernetes cluster on his machine.

2)

True, this is now a duplication of manifest yamls.
So if we want continue to support a provisioning of an environment (bare metal or virtual machines doesn't matter) we should probably find a solution for this (maybe with a submodule to the newly helm repo)

**Pull request merged**